playwright/Customizations/Compliance.spec.ts

```diff
@@ -83,3 +89,108 @@ test('Create a blueprint with Compliance policy selected', async ({
+  await cleanup.add(() => deleteCompliancePolicy(page, policyName));
+
+  await ensureAuthenticated(page);
+
+  await createCompliancePolicy(page, policyName, policyType);
+
+  await test.step('Wait for policy to be available', async () => {
+    await navigateToCompliancePolicy(page, policyName);
+    // Wait a bit longer for the policy to be fully available after creation
+    await expect(page.getByRole('row', { name: policyName })).toBeVisible({
+      timeout: 60000,
+    });
+  });
+
+  if (isHosted()) {
+    await page.goto('/insights/image-builder/imagewizard?release=rhel9');
+  } else {
+    await page.goto('/imageWizard?release=rhel9');
+  }
+  const frame = await ibFrame(page);
+
+  await test.step('Navigate to optional steps in Wizard', async () => {
+    // Wait for RHEL 9 to be selected (the wizard should open directly with RHEL 9)
+    await expect(frame.getByTestId('release_select')).toHaveText(
+      'Red Hat Enterprise Linux (RHEL) 9',
+      { timeout: 10000 },
+    );
+    await frame.getByRole('checkbox', { name: 'Virtualization' }).click();
+    await frame.getByRole('button', { name: 'Next' }).click();
+    await registerLater(frame);
+  });
+
+  await test.step('Select and fill the Compliance step', async () => {
+    await frame
+      .getByLabel('Wizard steps')
+      .getByRole('button', { name: 'Security' })
+      .click();
+    await frame
+      .getByRole('radio', { name: 'Use a custom compliance policy' })
+      .click();
+    await frame.getByRole('button', { name: 'None' }).click();
+    await expect(frame.getByRole('option').first()).toBeVisible();
+    const searchInput = frame.getByRole('textbox', { name: /filter/i });
+    await expect(searchInput).toBeVisible();
+    await searchInput.fill(policyName);
+    await expect(frame.getByRole('option', { name: policyName })).toBeVisible({
+      timeout: 30000,
+    });
+    await frame.getByRole('option', { name: policyName }).click();
+    await expect(frame.getByRole('button', { name: policyName })).toBeVisible();
+    await frame.getByRole('button', { name: 'Review and finish' }).click();
+  });
+
+  await test.step('Fill the BP details', async () => {
+    await fillInDetails(frame, blueprintName);
+  });
+
+  await test.step('Create BP', async () => {
+    await createBlueprint(frame, blueprintName);
+  });
+
+  await removeCompliancePolicyRule(
+    page,
+    policyName,
+    'Install firewalld Package',
+  );
+
+  await test.step('Verify compliance warning appears in blueprint', async () => {
+    await navigateToLandingPage(page);
+    const updatedFrame = await ibFrame(page);
+
+    const searchInput = updatedFrame.getByRole('textbox', {
+      name: 'Search input',
+    });
+    await expect(searchInput).toBeVisible();
+    await searchInput.fill(blueprintName);
+    await expect(searchInput).toHaveValue(blueprintName);
+
+    const blueprintButton = updatedFrame.locator(
+      `button[id="${blueprintName}"]`,
+    );
```

Comment:
If we can use a different locator than `locator`, that would be nice. I'd help here, but I currently do not have resources for testing this due to stage refresh.